SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -242,11 +242,19 @@ void USpatialNetDriver::InitiateConnectionToSpatialOS(const FURL& URL)
 	// If arguments can not be found we will use the regular flow of loading from the input URL.
 
 	FString SpatialWorkerType = GetGameInstance()->GetSpatialWorkerType().ToString();
-	if (!GameInstance->GetFirstConnectionToSpatialOSAttempted() && !GameInstance->GetPreventAutoConnectWithLocator())
+
+	if (!GameInstance->GetFirstConnectionToSpatialOSAttempted())
 	{
 		GameInstance->SetFirstConnectionToSpatialOSAttempted();
-		if (!ConnectionManager->TrySetupConnectionConfigFromCommandLine(SpatialWorkerType))
+		if (GetPreventAutoConnectWithCommandLineArgs())
+		{
+			// If first time connecting but the bPreventAutoConnectWithCommandLineArgs flag is set then use input URL to setup connection config.
```

Comment:
Could you explain this a little more - how does this prevent the connection from going ahead?

Reply:
It does not prevent the connection but it prevents a connection using command line arguments. If GetPreventAutoConnectWithCommandLineArgs return true we will useSetupConnectionConfigFromURL which does not use command line args. Otherwise we would try TrySetupConnectionConfigFromCommandLine. 

GetPreventAutoConnectWithCommandLineArgs itself checks to see if we are a client and the flag is set. If we are a server GetPreventAutoConnectWithCommandLineArgs will always return false.

Reply:
So what happens if we set this value and run in PIE?

Reply:
Hmm true... the check for PIE should be done in this function too. I have it in the NetDriver.